mmtk/src/abi.rs

```diff
@@ -168,7 +172,7 @@ impl InstanceKlass {
     const VTABLE_START_OFFSET: usize = Self::HEADER_SIZE * BYTES_IN_WORD;
 
     fn start_of_vtable(&self) -> *const usize {
-        unsafe { (self as *const _ as *const u8).add(Self::VTABLE_START_OFFSET) as _ }
+        unsafe { (self as *const Self as *const u8).add(Self::VTABLE_START_OFFSET) as _ }
```

Comment:
Should be easier to use `Address` here, too.  The function `start_of_itable` below can be changed similarly, too.

```suggestion
        (Address::from_ref(self) + Self::VTABLE_START_OFFSET).to_ptr()
```

Reply:
Fixed